### ISSUE TITLE
Service Data Download leads to API request fail

### ISSUE THREAD

**Issue by zauster** [ropensci/comtradr#5]
Hey, 

nice package! 
I have troubles downloading service trade data (on a fresh started R instance):

```
> library(comtradr)
> this.year <- 2015
> tmp <- ct_search(reporter = "Austria",
+                  partners = "World", freq = "annual",
+                  start_date = as.Date(paste0(this.year, "/01/01")),
+                  end_date = as.Date(paste0(this.year, "/12/31")),
+                  type = "services")
Error: API request failed, with status code [200]
Fail Reason: Invalid classification for trade type.
> sessionInfo()
R version 3.4.2 (2017-09-28)
Platform: x86_64-w64-mingw32/x64 (64-bit)
Running under: Windows 7 x64 (build 7601) Service Pack 1

Matrix products: default

locale:
[1] LC_COLLATE=English_United Kingdom.1252 
[2] LC_CTYPE=English_United Kingdom.1252   
[3] LC_MONETARY=English_United Kingdom.1252
[4] LC_NUMERIC=C                           
[5] LC_TIME=English_United Kingdom.1252    

attached base packages:
[1] stats     graphics  grDevices utils     datasets  methods   base     

other attached packages:
[1] comtradr_0.1.0      ormisc_0.1          openxlsx_4.0.17    
[4] devtools_1.13.4     data.table_1.10.4-3

loaded via a namespace (and not attached):
 [1] Rcpp_0.12.14     digest_0.6.12    withr_2.1.0.9000 R6_2.2.2        
 [5] jsonlite_1.5     magrittr_1.5     httr_1.3.1       rlang_0.1.4     
 [9] curl_3.0         tools_3.4.2      purrr_0.2.4      compiler_3.4.2  
[13] memoise_1.1.0 
```

Any idea what's wrong?

**Comment by ChrisMuir** [comment]
Hi @zauster, thanks for bringing this to my attention! The error message indicates that there's an issue with the arg `type`.

I honestly haven't tested the functions as well as I should have for param `type = "services"`, and this example exposed a bunch of holes in how the package currently handles it. I'll go through the issues one by one, and will end up opening separate issues for some of these:

1. Performing a query with `type = "services"` *requires* that the commodity db scheme be "EB02"...this was news to me! This is clear as day if you go to the Comtrade data [web interface](https://comtrade.un.org/data/), select "Services" under "Type of product", and the only option for variable Classification is "EBOPS".

I'll be adding a check for this early within `ct_search`, so that it's able to throw a much more helpful error message. The default commodity db file that ships with the package is "HS", and the way to download the correct file for "services" is to run `comtradr::ct_update_databases("EB02")`. If you take that step prior to running your query, the error message goes away, however it still always returns a zero row df (explained in number 2 below).

2. In `ct_search`, the default value for the commodity codes arg is `TOTAL`, which is meant to mean "all commodities". In the guts of `ct_search`, the literal string `TOTAL` ends up getting used within the query url...this works for all of the different "HS" commodity schemes, however for "EB02", "TOTAL - all services" has an actual commodity code of `200`.

To fix this within `ct_search`, when "all" or "total" is passed as input to arg `commod_codes`, I'll have a step that actually looks up the code for either in the current commodity codes db on file. For any of the "HS" commodity schemes, this would return "ALL" or "TOTAL", but for "EB02" this would return "ALL" or "200".

3. In your example, the error message is this:
```
Error: API request failed, with status code [200]
Fail Reason: Invalid classification for trade type.
```
The bit about "invalid classification" comes straight from the Comtrade API. Basically anytime a request returns no data AND includes a message in a specific field of the request headers, this is an indication that there was some input parameter error (and the message is supposed to say what caused the error). So the pkg just takes the API message and includes it as part of the error message the user sees (which is good), and it also includes info on the status code....however, I think I'm going to remove the language referencing the status code in these situations. I think it's misleading, at first glance it looks like there was a status code issue.

I want to say thank you again @zauster for taking the time to open this issue, there are a number of different fixes I need to work on. This will end up making the package that much better...so thanks!

**Comment by ChrisMuir** [comment]
Oh and also, while working through these issues, I found this bug:

```r
ct_commodity_lookup("halibut")
#> $halibut
#> [1] "030221 - Fish; halibut (reinhardtius hippoglossoides, hippoglossus hippoglossus, hippoglossus stenolepis), fresh or chilled (excluding fillets, livers, roes and other fish meat of heading no. 0304)"
#> [2] "030331 - Fish; halibut (reinhardtius hippoglossoides, hippoglossus hippoglossus, hippoglossus stenolepis), frozen (excluding fillets, livers, roes and other fish meat of heading no. 0304)"

ct_commodity_lookup("HALIBUT")
#> $HALIBUT
#> character(0)

#> Warning message:
#> There were no matching results found 
```
These two commodity lookup queries should return the same results.

**Comment by ChrisMuir** [comment]
This issue has been fixed via commit [cd25402](https://github.com/ropensci/comtradr/commit/cd254021d3fd207877330db950470ee774071bb3).

Your example query now outputs a data frame with two observations (one for imports, one for exports, both obs for 2015).

```r
library(comtradr)
this.year <- 2015
tmp <- ct_search(reporter = "Austria", 
                 partners = "World", 
                 freq = "annual", 
                 start_date = as.Date(paste0(this.year, "/01/01")), 
                 end_date = as.Date(paste0(this.year, "/12/31")), 
                 type = "services")
nrow(tmp)
#> 2
```
Thank you again @zauster for bringing this to my attention, this allowed me to fix three bugs and make an improvement to error messaging!

**Comment by zauster** [comment]
Thanks a lot Chris for these fast reponses/fixes! I will try them out first thing tomorrow...
